src/redux/reducers.ts

```diff
@@ -50,7 +50,7 @@ export const gameReducer = (
             const {id, answer} = action.payload
             const exercise = state.catalogue.exercises
                 .filter(e => e.id === id)
-                .map(e => Object.assign({}, e, { correct: correctAnswer(e, answer) }))[0];
+                .map(e => Object.assign({}, e, { correct: correctAnswer(e,answer) }))[0];
```

Comment:
doesn't use the new object structure?  also this is doing the both lookup and modify at the same time that breaks the command & query principle.